src/DataLayouts/DataLayouts.jl

```diff
@@ -92,6 +92,41 @@ abstract type AbstractDispatchToDevice end
 struct ToCPU <: AbstractDispatchToDevice end
 struct ToCUDA <: AbstractDispatchToDevice end
 
+"""
+    AbstractMask
+
+An abstract mask type, for marking domains as present or absent.
+"""
+abstract type AbstractMask end
+
+"""
+    NoMask
+
+A lazy non-mask type (default), used to indicate that an entire domain is
+present.
+"""
+struct NoMask <: AbstractMask end
```

Comment:
Could you explain why we have both enable_mask and NoMask? It seems like then we have three options:'
- no mask enabled
- mask enabled but NoMask used (is this one needed?)
- mask enabled and mask used

Reply:
`enable_mask` is user-facing, and turns on a mask for the space. `NoMask` (an internal object) is the default (when `enable_mask = false`), and `IJHMask` (also an internal object) is used when `enable_mask = true` for the appropriate space. If we want to support `SpectralElementSpace1D`, then we'll need to define `IHMask`.